hubstorage/resourcetype.py

```diff
@@ -65,24 +90,24 @@ def _add_resume_param(self, lastline, offset, params):
     def iter_values(self, *args, **kwargs):
         """Reliably iterate through all data as python objects
 
-        calls iter_json, decoding the results
+        calls either iter_json or iter_msgpack, decoding the results
         """
+        if self._supports_msgpack:
+            return mpdecode(self.iter_msgpack(*args, **kwargs))
         return jldecode(self.iter_json(*args, **kwargs))
 
-    def iter_json(self, _path=None, requests_params=None, **apiparams):
-        """Reliably iterate through all data as json strings"""
-        requests_params = dict(requests_params or {})
-        requests_params.setdefault('method', 'GET')
-        requests_params.setdefault('stream', True)
+    def retry(self, iter_callback, resume=False, _path=None, requests_params=None, **apiparams):
```

Comment:
IMO shouldn't be public. and just in case - didn't you think of using [retrying](https://pypi.python.org/pypi/retrying) lib? e.g. decorate ._iter_content and ._iter_lines with it?